### PR TITLE
Remove char escape from string template

### LINKED ISSUE #23988: Remove char escape from String template
Example: https://github.com/ballerina-platform/ballerina-lang/blob/4a2a71fffa167992843dfb43e7b70a6b79912731/tests/jballerina-unit-test/src/test/resources/test-src/types/string/string-template-literal.bal#L4

In the above file, we have used `\` to escape some characters, But that is not supported in the Spec.

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #23988 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -3916,7 +3916,9 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 position = matcher.end() - 2;
                 matcher = UNICODE_PATTERN.matcher(text);
             }
-            text = StringEscapeUtils.unescapeJava(text);
+            if (type != SyntaxKind.TEMPLATE_STRING) {
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3919]: This applies to XML literal as well.

Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Done

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 2]
```diff
@@ -3916,7 +3916,9 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 position = matcher.end() - 2;
                 matcher = UNICODE_PATTERN.matcher(text);
             }
-            text = StringEscapeUtils.unescapeJava(text);
+            if (type != SyntaxKind.TEMPLATE_STRING && type != SyntaxKind.XML_TEXT_CONTENT) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:3919]: Need to consider raw templates here as well

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/XMLQueryExpressionTest.java [hunk 3]
```diff
@@ -48,13 +48,13 @@ public void testSimpleQueryExprForXML() {
...
 
-    @Test(description = "Test simple query expression for XMLs - #2")
+    @Test(groups = { "disableOnOldParser" }, description = "Test simple query expression for XMLs - #2")
     public void testSimpleQueryExprForXML2() {
         BValue[] returnValues = BRunUtil.invoke(result, "testSimpleQueryExprForXML2");
         Assert.assertNotNull(returnValues);
 
         Assert.assertEquals(returnValues[0].stringValue(),
-                "<book>the book</book>bit of text✂✅");
+                "<book>the book</book>bit of text\\u2702\\u2705");
```
Comment (rdhananjaya) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/XMLQueryExpressionTest.java:57]: I think this is correct.
Spec [1] says:  Within a BacktickString, every character that is not part of an interpolation is interpreted as a literal character. 

[1] - https://ballerina.io/ballerina-spec/spec.html#section_6.6.1

**Comment by pubudu91** [comment]
@Chirantha Can you sync this with master?

**Pull request merged**